### PR TITLE
GAWB-913: Add library curator role and related endpoints

### PR DESCRIPTION
**Pull request by MatthewBemis**
I've created the following three Google groups to go along with this:

```
fc-CURATORS@dev.test.firecloud.org
fc-CURATORS@alpha.test.firecloud.org
fc-CURATORS@staging.test.firecloud.org
```

~~TODO: firecloud-admin-cli updates~~
- [x] Add a note about the new Google group to the release doc
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate. 
  - If doing a migration, take a backup of the dev and alpha DBs in Google Cloud Console
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: If PR includes new or changed db queries, include the explain plans in the description
- [x] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell  that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by ansingh7115** [comment]
👍

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala [hunk 1]
```diff
@@ -778,6 +802,7 @@ class HttpGoogleServicesDAO(
   def toGoogleGroupName(groupName: RawlsGroupName) = s"GROUP_${groupName.value}@${appsDomain}"
 
   def adminGroupName = s"${groupsPrefix}-ADMINS@${appsDomain}"
+  def curatorGroupName = s"${groupsPrefix}-CURATORS@${appsDomain}"
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala:805]: when is this group created? Should we create it automatically if it does not exist?

File: src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 2]
```diff
@@ -127,6 +131,10 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
 
     case AdminDeleteRefreshToken(userRef) => asFCAdmin { deleteRefreshToken(userRef) } pipeTo sender
     case AdminDeleteAllRefreshTokens => asFCAdmin { deleteAllRefreshTokens() } pipeTo sender
+
+    case IsLibraryCurator(userEmail) => { isLibraryCurator(userEmail) } pipeTo sender
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala]: why isn't this admin too?

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala (hunk 1 above)
Comment (MatthewBemis) [reply src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala:805]: I created it manually in all envs except for prod. I don't think this group necessarily deserves different treatment than groups like fc-ADMINS and dbGapAuthorizedUsers but if we want to start a trend of doing these things automatically I'm fine with that

**Pull request merged**